scripts/src/java/org/oppia/android/scripts/testing/TestBazelWorkspace.kt

```diff
@@ -0,0 +1,231 @@
+      writer.println("    name = \"$testName\",")
+      writer.println("    srcs = [\"${testFile.name}\"],")
+      writer.print("    deps = [")
+      if (withGeneratedDependency) {
+        testDependencyNameMap[testName] = dependencyTargetName ?: error("Something went wrong.")
+        writer.print("\":$dependencyTargetName\",")
+      }
+      withExtraDependency?.let { writer.print("\"$it\",") }
+      writer.println("],")
+      writer.println(")")
+      return@use listOf(testFile, buildFile) + prereqFiles + libPrereqFiles
+    }.toSet()
+  }
+
+  /**
+   * Generates a new test target using the specified [testName], and generating a local test file
+   * using the test name as the filename (with a Kotlin extension).
+   *
+   * For details on the parameters & return value, see [addTestToBuildFile] for more context.
+   */
+  fun createTest(
+    testName: String,
+    withGeneratedDependency: Boolean = false,
+    withExtraDependency: String? = null,
+    subpackage: String? = null
+  ): Iterable<File> {
+    check(testName !in testFileMap) { "Test '$testName' already exists" }
+    val testFile = if (subpackage != null) {
+      if (!File(temporaryRootFolder.root, subpackage).exists()) {
+        temporaryRootFolder.newFolder(subpackage)
+      }
+      temporaryRootFolder.newFile("$subpackage/$testName.kt")
+    } else temporaryRootFolder.newFile("$testName.kt")
+    return addTestToBuildFile(
+      testName,
+      testFile,
+      withGeneratedDependency,
+      withExtraDependency,
+      subpackage
+    )
+  }
+
+  /**
+   * Generates a new library using the specified dependency name, modifying the local
+   * [rootBuildFile] and updating the WORKSPACE to support Kotlin libraries, if necessary.
+   *
+   * @param dependencyName the name of the library dependency, which will be used both for the
+   *     target name and a local source file representing the 'library'. For example, if "SampleLib"
+   *     is passed as the dependency name then a "SampleLib.kt" will be generated and the library's
+   *     target name will be "SampleLib_lib".
+   * @return a pair where the first value is the library's target name and the second value is an
+   *     iterable of files that were changed as part of generating this library
+   */
+  fun createLibrary(dependencyName: String): Pair<String, Iterable<File>> {
+    val libTargetName = "${dependencyName}_lib"
+    check(libTargetName !in libraryFileMap) { "Library '$dependencyName' already exists" }
+    val prereqFiles = ensureWorkspaceIsConfiguredForKotlin()
+    prepareBuildFileForLibraries(rootBuildFile)
+    return rootBuildFile.appendingPrintWriter().use { writer ->
+      val depFile = temporaryRootFolder.newFile("$dependencyName.kt")
+      libraryFileMap[libTargetName] = depFile
+      writer.println("kt_jvm_library(")
+      writer.println("    name = \"$libTargetName\",")
+      writer.println("    srcs = [\"${depFile.name}\"],")
+      writer.println(")")
```

Comment:
```kotlin
val ktJvmLibrary = 
"""
kt_jvm_library(
    name = "$libTargetName",
    srcs = ["${depFile.name}"],
)
"""
writer.println(ktJvmLibrary)
```
This might look a bit neat. Ditto elsewhere.

Reply:
Nice idea--this also let me get rid of the ``appendingPrintWriter`` helper function.